query-node/mappings/src/bounty.ts

```diff
@@ -0,0 +1,485 @@
+  await store.save<BountyEntry>(entry)
+
+  // Record the event
+  const announcedEvent = new WorkEntryAnnouncedEvent({ ...genericEventFields(event), entry })
+  await store.save<WorkEntryAnnouncedEvent>(announcedEvent)
+}
+
+// Change withdrawn entries status to withdrawn
+export async function bounty_WorkEntryWithdrawn({ event, store }: EventContext & StoreContext): Promise<void> {
+  const entryWithdrawnEvent = new BountyEvents.WorkEntryWithdrawnEvent(event)
+
+  // Update the entry status
+  const entry = await updateEntry(store, event, entryWithdrawnEvent.params[1], () => ({
+    status: new BountyEntryStatusWithdrawn(),
+  }))
+
+  // Record the event
+  const withdrawnInEvent = new WorkEntryWithdrawnEvent({ ...genericEventFields(event), entry })
+  await store.save<WorkEntryWithdrawnEvent>(withdrawnInEvent)
+}
+
+// Store WorkEntrySlashed events
+export async function bounty_WorkEntrySlashed({ event, store }: EventContext & StoreContext): Promise<void> {
+  const entrySlashedEvent = new BountyEvents.WorkEntrySlashedEvent(event)
+  const [, entryId] = entrySlashedEvent.params
+
+  // Record the event
+  const entry = new BountyEntry({ id: String(entryId) })
+  const slashedInEvent = new WorkEntrySlashedEvent({ ...genericEventFields(event), entry })
+  await store.save<WorkEntrySlashedEvent>(slashedInEvent)
+}
+
+// Store WorkSubmitted events
+export async function bounty_WorkSubmitted({ event, store }: EventContext & StoreContext): Promise<void> {
+  const entrySlashedEvent = new BountyEvents.WorkEntrySlashedEvent(event)
+
+  // Update the entry status
+  const entry = await updateEntry(store, event, entrySlashedEvent.params[1], () => ({
+    workSubmitted: true,
+  }))
+
+  // Record the event
+  const submittedInEvent = new WorkSubmittedEvent({ ...genericEventFields(event), entry })
+  await store.save<WorkEntrySlashedEvent>(submittedInEvent)
+}
+
+// Start bounties withdrawal period and set entries status to either passed, winner, or rejected
+export async function bounty_OracleJudgmentSubmitted({ event, store }: EventContext & StoreContext): Promise<void> {
+  const judgmentSubmittedEvent = new BountyEvents.OracleJudgmentSubmittedEvent(event)
+  const [bountyId, , bountyJudgment] = judgmentSubmittedEvent.params
+
+  const entryJudgments = Array.from(bountyJudgment.entries())
+
+  // Update the bounty status
+  const hasWinners = entryJudgments.some(([, judgment]) => judgment.isWinner)
+  const bounty = await updateBounty(store, event, bountyId, () => ({
+    stage: BountyStage[hasWinners ? 'Successful' : 'Failed'],
+  }))
+
+  // Update winner entries status
+  await Promise.all(
+    bounty.entries?.map((entry) => {
+      const judgement = entryJudgments.find(([entryId]) => String(entryId) === entry.id)?.[1]
+
+      if (!judgement) {
+        entry.status = new BountyEntryStatusPassed()
+      } else if (judgement?.isWinner) {
+        const status = new BountyEntryStatusWinner()
+        status.reward = judgement.asWinner.reward
```

Comment:
Due to Hydra bug `u128` cannot be directly assigned to variant field (as mentioned before)
```suggestion
        status.reward = new BN(judgement.asWinner.reward.toString())
```